app/models/enterprise.rb

```diff
@@ -187,8 +177,25 @@ def self.distinct_count
     count(distinct: true)
   end
 
+  def contact
+    manager = EnterpriseRole.find_by_enterprise_id_and_receives_notifications(self.id, true)
+    return owner if manager.blank?
+    contact = Spree::User.find(manager.user_id)
+    contact || owner
+  end
+
+  def update_contact(user_id)
+    transaction do
+      current_contact = EnterpriseRole.find_by_receives_notifications_and_enterprise_id true, id
+      current_contact.update_attribute :receives_notifications, false if current_contact
+
+      new_contact = EnterpriseRole.find_or_create_by_user_id_and_enterprise_id user_id.to_i, id
+      new_contact.update_attribute :receives_notifications, true
+    end
+  end
```

Comment:
This didn't work?:

````ruby
# enterprise.rb
def set_contact(user_id)
  # maybe raise an error here unless user_id is actually a manager?
  enterprise_roles.update_all(["receives_notifications=(user_id=?)",user_id])
end
````

One query instead of 4...